packages/embedded-utils/src/utils/handlePostMessage.ts

```diff
@@ -3,19 +3,23 @@ import { ISpace } from '../types'
 
 export const handlePostMessage = (
   closeSpace: ISpace['closeSpace'],
-  listener: FlatfileListener
+  listener: FlatfileListener,
+  onClose?: () => void
 ) => {
   return (message: MessageEvent<{ flatfileEvent: FlatfileEvent }>) => {
     const { flatfileEvent } = message.data
-    if (!flatfileEvent) return
+    if (!flatfileEvent) {
+      return
+    }
     if (
       flatfileEvent.topic === 'job:outcome-acknowledged' &&
       flatfileEvent.payload.status === 'complete' &&
       flatfileEvent.payload.operation === closeSpace?.operation &&
       closeSpace &&
       typeof closeSpace.onClose === 'function'
     ) {
-      closeSpace.onClose({event: flatfileEvent})
+      closeSpace.onClose({ event: flatfileEvent })
+      onClose?.()
     }
     listener.dispatchEvent(flatfileEvent)
```

Comment:
If we're going to use this for both listening for all messages and closing the portal instance/cleaning up listeners, do we want to not `listener.dispatchEvent(flatfileEvent)` an event onClose?

Reply:
It shouldn't matter right, they may still want to listen for the event normally, this is still a standard `'job:outcome-acknowledged'` event that they could do something with, even thought it would be duplicated. For example in `apps/vanilla` we're logging all events to the console, and removing this dispatchEvent would prevent seeing the acknowledge job.